Adapt CAConstants for HIon

stahlleiton opened the pull request:
This PR include changes needed to avoid overflow while running the patatrack pixel tracking on HIon HLT menu using 2018 PbPb data.

The changes increase the thresholds of the following CA constants:

- maxNumberOfTuples : increased from 24 * 1024   to    32 * 1024 to avoid overflow on number of tuples.
- maxCellsPerHit : increased from 128   to    2 * 128 to avoid overflow on OuterHitOfCell.

#### PR validation:

The changes were tested on 2018 PbPb data using the HardProbe PDs.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

@fwyzard

jshlee committed:
removed obsolete classes
changed classes to consistent structure and naming

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-34811/24555

- This PR adds an extra 12KB to repository

cmsbuild commented:
A new Pull Request was created by  @stahlleiton (Andre Stahl) for master.

It involves the following packages:

- RecoPixelVertexing/PixelTriplets (**reconstruction**)


@perrotta, @jpata, @cmsbuild, @slava77 can you please review it and eventually sign? Thanks.
@felicepantaleo, @GiacomoSguazzoni, @JanFSchulte, @rovere, @VinInn, @mmusich, @mtosi, @dgulhan this is something you requested to watch as well.
@silviodonato, @dpiparo, @qliphy, @perrotta you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

fwyzard commented on RecoPixelVertexing/PixelTriplets/plugins/CAConstants.h:
```diff
@@ -32,9 +32,9 @@ namespace caConstants {
 #else   // GPU_SMALL_EVENTS
   // tested on MC events with 55-75 pileup events
-  constexpr uint32_t maxNumberOfTuples = 24 * 1024;
+  constexpr uint32_t maxNumberOfTuples = 32 * 1024;
```
```suggestion
  // and extended for Heavy Ions operations (24k -> 32k tuples, 128 -> 256 cells)
  constexpr uint32_t maxNumberOfTuples = 32 * 1024;
```

fwyzard commented:
+heterogeneous

fwyzard commented:
please test

fwyzard commented:
enable gpu

fwyzard commented:
please test

fwyzard commented:
assign heterogeneous

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-34811/24556

- This PR adds an extra 12KB to repository

cmsbuild commented:
New categories assigned: heterogeneous

@fwyzard,@makortel you have been requested to review this Pull request/Issue and eventually sign? Thanks

cmsbuild commented:
Pull request #34811 was updated. @perrotta, @makortel, @fwyzard, @jpata, @slava77 can you please check and sign again.

cmsbuild commented:
-1

**Failed Tests**: RelVals-INPUT 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/summary.html
**COMMIT**: aabb0db56fa03edd171b27202797bc44d1a96c45
**CMSSW**: CMSSW_12_1_X_2021-08-05-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/34811/17611/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals-INPUT

- [11601.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11601.0_SingleElectronPt10+2021+SingleElectronPt10_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA)```11601.0_SingleElectronPt10+2021+SingleElectronPt10_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA/step2_SingleElectronPt10+2021+SingleElectronPt10_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA.log```
- [11602.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11602.0_SingleElectronPt35+2021+SingleElectronPt35_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA)```11602.0_SingleElectronPt35+2021+SingleElectronPt35_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA/step2_SingleElectronPt35+2021+SingleElectronPt35_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA.log```
- [11603.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11603.0_SingleElectronPt1000+2021+SingleElectronPt1000_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA)```11603.0_SingleElectronPt1000+2021+SingleElectronPt1000_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA/step2_SingleElectronPt1000+2021+SingleElectronPt1000_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [11604.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11604.0_SingleGammaPt10+2021+SingleGammaPt10_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11605.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11605.0_SingleGammaPt35+2021+SingleGammaPt35_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11606.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11606.0_SingleMuPt1+2021+SingleMuPt1_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11607.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11607.0_SingleMuPt10+2021+SingleMuPt10_Eta2p85_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11608.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11608.0_SingleMuPt100+2021+SingleMuPt100_Eta2p85_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11609.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11609.0_SingleMuPt1000+2021+SingleMuPt1000_Eta2p85_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11630.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11630.0_QCD_Pt_3000_3500_14TeV+2021+QCD_Pt_3000_3500_14TeV_TuneCUETP8M1_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11634.0_TTbar_14TeV+2021+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11634.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11634.1_TTbar_14TeV+2021_trackingOnly+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.24](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11634.24_TTbar_14TeV+2021_0T+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11634.5](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11634.5_TTbar_14TeV+2021_pixelTrackingOnly+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.501](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11634.501_TTbar_14TeV+2021_Patatrack_PixelOnlyCPU+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.505](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11634.505_TTbar_14TeV+2021_Patatrack_PixelOnlyTripletsCPU+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.521](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11634.521_TTbar_14TeV+2021_Patatrack_HCALOnlyCPU+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.511](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11634.511_TTbar_14TeV+2021_Patatrack_ECALOnlyCPU+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.7](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11634.7_TTbar_14TeV+2021_trackingMkFit+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.91](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11634.91_TTbar_14TeV+2021_trackdnn+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11640.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11640.0_MinBias_14TeV+2021+MinBias_14TeV_pythia8_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11643.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11643.0_QCD_FlatPt_15_3000HS_14+2021+QCDForPF_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11646.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11646.0_ZEE_14+2021+ZEE_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11650.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11650.0_ZMM_14+2021+ZMM_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11650.501](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11650.501_ZMM_14+2021_Patatrack_PixelOnlyCPU+ZMM_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11650.505](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11650.505_ZMM_14+2021_Patatrack_PixelOnlyTripletsCPU+ZMM_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11723.17](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11723.17_QCD_Pt_1800_2400_14+2021_seedingDeepCore+QCD_Pt_1800_2400_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11725.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11725.0_GluGluTo2Jets_14TeV+2021+GluGluTo2Jets_M_300_2000_14TeV_Exhume_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11834.0_TTbar_14TeV+2021PU+TTbar_14TeV_TuneCP5_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [11834.13](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11834.13_TTbar_14TeV+2021PU_mlpf+TTbar_14TeV_TuneCP5_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [11834.24](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11834.24_TTbar_14TeV+2021PU_0T+TTbar_14TeV_TuneCP5_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [11846.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11846.0_ZEE_14+2021PU+ZEE_14TeV_TuneCP5_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [12034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/12034.0_TTbar_14TeV+2021Design+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11925.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/11925.0_GluGluTo2Jets_14TeV+2021PU+GluGluTo2Jets_M_300_2000_14TeV_Exhume_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [12434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/12434.0_TTbar_14TeV+2023+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [12634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/12634.0_TTbar_14TeV+2023PU+TTbar_14TeV_TuneCP5_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [12834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/12834.0_TTbar_14TeV+2024+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [13034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/13034.0_TTbar_14TeV+2024PU+TTbar_14TeV_TuneCP5_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [23434.99](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/23434.99_TTbar_14TeV+2026D49PU_PMXS1S2+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14INPUT+PREMIX_PremixHLBeamSpot14PU+DigiTriggerPU+RecoGlobalPU+HARVESTGlobalPU)
- [23434.21](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/23434.21_TTbar_14TeV+2026D49PU_ProdLike+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14INPUT+DigiTriggerPU+RecoGlobalPU+MiniAODPU)
- [23434.9921](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/23434.9921_TTbar_14TeV+2026D49PU_PMXS1S2ProdLike+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14INPUT+PREMIX_PremixHLBeamSpot14PU+DigiTriggerPU+RecoGlobalPU+MiniAODPU)
- [34834.21](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/34834.21_TTbar_14TeV+2026D76PU_ProdLike+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14INPUT+DigiTriggerPU+RecoGlobalPU+MiniAODPU)
- [34834.9921](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17611/runTheMatrixINPUT-results/34834.9921_TTbar_14TeV+2026D76PU_PMXS1S2ProdLike+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14INPUT+PREMIX_PremixHLBeamSpot14PU+DigiTriggerPU+RecoGlobalPU+MiniAODPU)
</details>



## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 2 differences found in the comparisons
 - DQMHistoTests: Total files compared: 39
 - DQMHistoTests: Total histograms compared: 2999410
 - DQMHistoTests: Total failures: 7
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2999381
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 38 files compared)
 - Checked 165 log files, 37 edm output root files, 39 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_1_X_2021-08-05-2300+45810e/44656/triggerResults): no differences found

slava77 commented:
@cmsbuild please test

it looks like the gpu tests never went through

fwyzard commented:
From a simple test (building pixel triplets, GPU-only, pileup 50) this has a small but measurable impact.
```
2 CPUs:
  0: Intel(R) Xeon(R) Gold 6130 CPU @ 2.10GHz (16 cores, 32 threads)
  1: Intel(R) Xeon(R) Gold 6130 CPU @ 2.10GHz (16 cores, 32 threads)

1 visible NVIDIA GPUs:
  0: Tesla T4

Running 3 times over 10100 events with 2 jobs, each with 10 threads, 10 streams and 1 GPUs
```
  - reference: `1269.8 ±   3.6 ev/s`
  - with this PR: `1256.7 ±   4.0 ev/s`
  - impact: **`-1.0% ± 0.4%`**

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-45810e/17627/summary.html
**COMMIT**: aabb0db56fa03edd171b27202797bc44d1a96c45
**CMSSW**: CMSSW_12_1_X_2021-08-06-1100/slc7_amd64_gcc900
**Additional Tests**: GPU
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/34811/17627/install.sh` to create a dev area with all the needed externals and cmssw changes.


## GPU Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 4
 - DQMHistoTests: Total histograms compared: 19735
 - DQMHistoTests: Total failures: 72
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 19663
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 3 files compared)
 - Checked 12 log files, 9 edm output root files, 4 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsGPU/CMSSW_12_1_X_2021-08-06-1100+45810e/44665/triggerResults): no differences found


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 39
 - DQMHistoTests: Total histograms compared: 2999410
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2999387
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 38 files compared)
 - Checked 165 log files, 37 edm output root files, 39 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_1_X_2021-08-06-1100+45810e/44667/triggerResults): no differences found

stahlleiton commented:
> From a simple test (building pixel triplets, GPU-only, pileup 50) this has a small but measurable impact.
> 
> ```
> 2 CPUs:
>   0: Intel(R) Xeon(R) Gold 6130 CPU @ 2.10GHz (16 cores, 32 threads)
>   1: Intel(R) Xeon(R) Gold 6130 CPU @ 2.10GHz (16 cores, 32 threads)
> 
> 1 visible NVIDIA GPUs:
>   0: Tesla T4
> 
> Running 3 times over 10100 events with 2 jobs, each with 10 threads, 10 streams and 1 GPUs
> ```
> 
>     * reference: `1269.8 ±   3.6 ev/s`
> 
>     * with this PR: `1256.7 ±   4.0 ev/s`
> 
>     * impact: **`-1.0% ± 0.4%`**

Thanks for the information.
Is it also possible to test the change of only one parameter at the time to see which of the two parameters impose the largest impact on the throughput?

fwyzard commented:
The impact comes mostly from the change to the `maxCellsPerHit`:

constants | throughput
---|---
`maxNumberOfTuples = 24k`,  `maxCellsPerHit = 128` | `1269.8 ± 3.6 ev/s`
`maxNumberOfTuples = 24k`,  `maxCellsPerHit = 256` | `1258.9 ± 2.9 ev/s`
`maxNumberOfTuples = 32k`,  `maxCellsPerHit = 128` | `1265.2 ± 2.0 ev/s`
`maxNumberOfTuples = 32k`,  `maxCellsPerHit = 256` | `1256.7 ± 4.0 ev/s`

stahlleiton commented:
Thanks @fwyzard .
Then one option to reduce the impact could be to try to template only maxCellsPerHit .

fwyzard commented:
@VinInn, looking at [RecoPixelVertexing/PixelTriplets/plugins/gpuFishbone.h](https://github.com/cms-sw/cmssw/blob/master/RecoPixelVertexing/PixelTriplets/plugins/gpuFishbone.h), I'm wondering if the first part of the loop (the one that fills `x`, `y`, `z`, *etc.) can also be parallelised along `blockDim.x`, and those arrays moved into shared memory ?

If I followed the code correctly, even increasing `maxCellsPerHit` to `256` their combined size should be 22 bytes (4 floats, 1 int, 1 short) x 256 elements x 8 (blockDim.y) = 44k, which should fit in shared memory.

We may need to use cooperative groups to synchornise the sub-groups, but we can give it a try.

jshlee committed:
code-checks

smuzaffar committed:
added plugins/.cc files

smuzaffar committed:
Coderule script: fixes needed for python3

Commit:
Modify material budget code relevant to study XML vs DB

Commit:
Modified the examination of geometry in XML and DB

Commit:
Code check

Commit:
Code check

slava77 committed:
initialize Trackster::iterationIndex_

cmsbuild committed:
Merge pull request #34855 from smuzaffar/coderule-py3

Coderule script: fixes needed for python3

cmsbuild committed:
Merge pull request #34849 from smuzaffar/SimCalorimetry-fix-hdr

[SimCalorimetry] Avoid using private CondFormats/EcalObjects/src/classes.h

cmsbuild committed:
Merge pull request #34842 from bsunanda/Run3-sim100

Run3-sim100 Take Vladimir's comments for include statements of CLHEP headers

cmsbuild committed:
Merge pull request #34826 from smuzaffar/boost-regex

Replace boost::regex with std::regex

cmsbuild committed:
Merge pull request #34852 from iarspider/patch-3

[GCC11] Fix warning: loop variable binds to a temporary

cmsbuild committed:
Merge pull request #34832 from mmusich/sistripCommmissioningRunsDQM_12_1_X

[RFC] add `commissioning_run` as DQM runType and use it in SiStrip online DQM client

cmsbuild committed:
Merge pull request #34771 from Dr15Jones/fixHeadersMaker

Make header files in FWCore/Framework/interface consistent

cmsbuild committed:
Merge pull request #34837 from Dr15Jones/fasterMonitorElementData_Path_set

Performance improvements for DQM startup

cmsbuild committed:
Merge pull request #34856 from bsunanda/Run3-gex87

Run3-gex87 Modify material budget code relevant to study XML vs DB

lecriste committed:
Header Consistency

yeckang committed:
fixing wrong key for the histogram

smuzaffar committed:
[DQM] Remove unnecessary include statements to avoid private header dep

smuzaffar committed:
[L1TTrackMatch] Remove unnecessary include statements to avoid private header dep

lecriste committed:
Propagate review comments

smuzaffar committed:
[RecoMuon] Remove unnecessary include statements to avoid private hdr dep

smuzaffar committed:
[Reco] Removed unnecessary include statements

smuzaffar committed:
[Reco] Removed unnecessary include statements

smuzaffar committed:
[RecoPixelVertexing] Move public headers to interface directory

cmsbuild committed:
Merge pull request #34857 from bsunanda/Run3-sim101

Run3-sim101 Modified the examination of geometry in XML and DB

smuzaffar committed:
[RecoTracker] Move public headers to interface directory

smuzaffar committed:
added interface/getBestVertex.h

iarspider committed:
Add missing include in CondDBTools.cc

smuzaffar committed:
[TkTrackingRegions] Move public headers to interface directory

smuzaffar committed:
Update PhotonConversionTrajectorySeedProducerFromSingleLegAlgo.h

smuzaffar committed:
[Validation] Removed unnecessary include statements

smuzaffar committed:
[IOMC] Move public headers to interface directory

cmsbuild committed:
Merge pull request #34829 from bsunanda/Run3-alca196

Run3-alca196 Collapse .h and .cc files in DQMOffline/CalibCalo and reorganize the files

cmsbuild committed:
Merge pull request #34833 from perrotta/removeDeadAssignments

Remove dead assignments for ZToMuMuGammaAnalyzer and HGCalSimHitValidation

cmsbuild committed:
Merge pull request #34862 from smuzaffar/dqm-hdr-cleanup

[DQM] Remove unnecessary include statements to avoid private header dep

cmsbuild committed:
Merge pull request #34844 from CTPPS/pps-alignment-global-fix

PPS Alignment - bug fix

cmsbuild committed:
Merge pull request #34869 from iarspider/patch-8

Add missing include in CondDBTools.cc

smuzaffar committed:
[DBG] Fixes needed for 08-12-2300 DBG IB

kpedro88 committed:
more Python3 fixes for geometry script

mmusich committed:
fix miscelleanous mistakes in the all-in-one template configutations

mmusich committed:
introduce python compilation time check of the configuration files produced by the all-in-one tool

Commit:
Make P5 as the default for 2021 scenarios

smuzaffar committed:
code-format

cmsbuild committed:
Merge pull request #34871 from smuzaffar/Validation-hdr-fix

[Validation] Removed unnecessary include statements

cmsbuild committed:
Merge pull request #34870 from smuzaffar/TkTrackingRegions-hdr-fix

[TkTrackingRegions] Move public headers to interface directory

cmsbuild committed:
Merge pull request #34868 from smuzaffar/RecoTracker-hdr-fix

[RecoTracker] Move public headers to interface directory

cmsbuild committed:
Merge pull request #34867 from smuzaffar/RecoPixelVertexing-hdr-fix

[RecoPixelVertexing] Move public headers to interface directory

cmsbuild committed:
Merge pull request #34864 from smuzaffar/RecoMuon-fix-hdr

[RecoMuon] Remove unnecessary include statements to avoid private hdr deps

cmsbuild committed:
Merge pull request #34828 from mteroerd/ape_plots_and_bug_fix

Validation plots for APE and bug fixes

cmsbuild committed:
Merge pull request #34759 from ChrisMisan/from-CMSSW_12_0_0_pre2

Diamond Sampic reco

mmusich committed:
modernize LhcTrackAnalyzer

cmsbuild committed:
Merge pull request #34876 from mmusich/miscellaneousFixesAlignmentOfflineValidation_allInOneTool

[Tracker Alignment] Miscellaneous fixes in `Alignment/OfflineValidation` all-in-one tool

cmsbuild committed:
Merge pull request #34853 from smuzaffar/CalibMuon-merge-plugin

[CalibMuon] Move and merge .h and .cc files used by plugin

cmsbuild committed:
Merge pull request #34821 from yeckang/gemValid_12_0_0_pre3

GEM Validation update

cmsbuild committed:
Merge pull request #34505 from UniMiBAnalyses/BiancaPinolini-Ecal-reco-phase2

Adding the codes for ECAL Phase2 amplitude local reconstruction

kpedro88 committed:
make type fully consistent, avoid magic number

cmsbuild committed:
Merge pull request #34800 from lecriste/SimTS_fromAnyCP

[HGCAL] From CaloParticle to SimTrackster in the Validation linking section

lecriste committed:
Restore separation between hits calibration and validation

lecriste committed:
Remove duplictaed hardcoded label 'simClusters'

lecriste committed:
Remove duplictaed hardcoded label 'ClusterLevel'

lecriste committed:
Remove duplictaed hardcoded label 'hgcalLayerClusters'

Commit:
Put P4 in the list of depricated class

cmsbuild committed:
Merge pull request #34809 from guitargeek/deps_1

Clean unused dependencies from BuildFiles

cmsbuild committed:
Merge pull request #34881 from mmusich/modernizeLhcTrackAnalyzer

[Alignment/HIPAlignmentAlgorithm] modernize `LhcTrackAnalyzer`

smuzaffar committed:
Update RecoLocalCalo/HGCalRecProducers/plugins/HeterogeneousHGCalHEFCellPositionsConditions.cc

Co-authored-by: Slava Krutelyov <slava77@gmail.com>

cmsbuild committed:
Merge pull request #34875 from kpedro88/Phase2-WF74

more Python3 fixes for geometry script

smuzaffar committed:
Buildfile cleanup: remove non-existing module.cc file

mmusich committed:
migrate AlignmentProducer to event consumes

cmsbuild committed:
Merge pull request #34879 from bsunanda/Run3-gex88

Run3-gex88 Make P5 as the default for 2021 scenarios

smuzaffar committed:
BuildFile: use py3-qlalchemy

cmsbuild committed:
Merge pull request #34874 from smuzaffar/dbg-fix210812

[DBG] Fixes needed for 08-12-2300 DBG IB

cmsbuild committed:
Merge pull request #34846 from OzAmram/TemplateInterpolate_isFinite_fix_master

Fix rare crash in pixel template interpolation

slava77 commented:
@fwyzard @stahlleiton 
please clarify on the status of this PR. From the last message about a week ago it sounded as if more developments are coming here.
Thank you.

Dr15Jones committed:
Moved headers from src to interface in FWCore/Framework

These headers were being used by other packages.

tvami committed:
Fix typo for skiped

fwyzard commented:
hi @slava77, yes I think we want to explore a couple of other possibilities before making these changes.

smuzaffar committed:
[CommonTools] Move public headers in to interface directory

cmsbuild committed:
Merge pull request #34886 from smuzaffar/condtools-py3-qlalchemy

BuildFile: use py3-sqlalchemy

Dr15Jones committed:
Added ThreadHandoff class

smuzaffar committed:
[Generator] Move public headers in to interface directory

smuzaffar committed:
[Fireworks] Move public headers in to interface directory

smuzaffar committed:
[Calib] Move private header in to src directory

Dr15Jones committed:
Run G4 worker on its own dedicated thread

Dr15Jones committed:
applied code format

Dr15Jones committed:
Removed thread_local from RunManagerMTWorker

OscarMTProducer now guarantees the class methods are run on their
own dedicated thread.

Dr15Jones committed:
Be able to set worker thread stack size in configuration

mmusich committed:
do not produce NaNs in SiPixelActionExecutor

cmsbuild committed:
Merge pull request #34892 from tvami/CosmeticFixRelMons

Fix typo in the RelMon category

cmsbuild committed:
Merge pull request #34883 from smuzaffar/SimG4CMS-bf-cleanup

Buildfile cleanup: remove non-existing module.cc file

cmsbuild committed:
Merge pull request #34897 from smuzaffar/calib-hdr-fix0816

[Calib] Move private header in to src directory

cmsbuild committed:
Merge pull request #34896 from smuzaffar/fireworks-hdr-fix0816

[Fireworks] Move public headers in to interface directory

cmsbuild committed:
Merge pull request #34697 from Dr15Jones/FWCoreFramework_task_chains

Apply task chains to more code in FWCore/Framework

smuzaffar committed:
[GCC10] Use constexpr which fixes gcc10 build error

cmsbuild committed:
Merge pull request #34900 from mmusich/fixWf8.0comparisons

Do not produce NaNs in `SiPixelActionExecutor`

bainbrid committed:
Read ROOT files for Seed BDTs (newly converted from XML files)

cmsbuild committed:
Merge pull request #34834 from jeongeun/recoTauTag12X

Migrate module configuration in RecoTauTag to use default cfipython

smuzaffar committed:
[Validation] Move EcalDigis sources in to plugins dir

smuzaffar committed:
[Validation] USe correct header for ParametersDefinerForTP

cmsbuild committed:
Merge pull request #34903 from smuzaffar/gcc10-build-fix

[GCC10] Use constexpr which fixes gcc10/ASAN build error

smorovic committed:
new environment variable used to instruct source to start from a
specified lumisection. Replaces another unused environment variable
which was not used (new name used for backwards compatibility).

makortel committed:
Complete migration of TrackClassifier to esConsumes()

smuzaffar committed:
[Geometry] Move public headers in to interface dir

smuzaffar committed:
code-format

Dr15Jones committed:
Created setupSiteLocalConfig

This function should be used rather than setting up SiteLocalConfigService by hand.

apsallid committed:
make makeHGCalValidationPlots.py executable again

smuzaffar committed:
[HLTriggerOffline] Move Higgs sources in to plugins dir

smuzaffar committed:
[PhysicsTools] Move icc file used by public interface in to interface dir

makortel committed:
Migrate (Cosmic)ParametersDefinerForTP to esConsumes()

Changed the (Cosmic)ParametersDefinerForTP from ESProducts to helper classes.

Dr15Jones committed:
Set random number system in worker thread

cmsbuild committed:
Merge pull request #34909 from apsallid/makeexeagain

[HGCAL] make makeHGCalValidationPlots.py executable again

lecriste committed:
Parallelize SimpleValidation

Dr15Jones committed:
Consolidated GLBMuonAnalyzer into one file

The module was never actually registered with the plugin manager
as SealModule.cc was not compiled.
This was changed to remove an unneeded dependency on
FWCore/Services/src/SiteLocalConfigService.h.

cmsbuild committed:
Merge pull request #34894 from smuzaffar/Generator-hdr-fix0816

[Generator] Move public headers in to interface directory

Dr15Jones committed:
Move Guid.h to interface

smuzaffar committed:
[Reco] Fixes needed to avoid exposing private headers via interface dir

cmsbuild committed:
Merge pull request #34893 from smuzaffar/CommonTools-fix-hdr

[CommonTools] Move public headers in to interface directory

tvami committed:
Add HCAL tag HcalRespCorrs_2021_v3.0_mc

makortel committed:
Remove unnecessary dependencies to boost and root

makortel committed:
Remove TrackAssociatorRecord as obsolete

makortel committed:
Remove MuonToSimAssociatorBase and MuonToSimAssociatorByHits as obsolete

These could have been removed in #8635. The testAssociatorRecoMuon has
been dysfuntional since then.

cmsbuild committed:
Merge pull request #34915 from Dr15Jones/setupSiteLocalConfig

Created setupSiteLocalConfig

cmsbuild committed:
Merge pull request #34891 from Dr15Jones/moreHeaderMovesFWCoreFramework

Moved headers from src to interface in FWCore/Framework

cmsbuild committed:
Merge pull request #34905 from smuzaffar/validation-correct-hdr0817

[Validation] Use correct header for ParametersDefinerForTP

cmsbuild committed:
Merge pull request #34911 from smuzaffar/HLTriggerOffline-hdr-fix0817

[HLTriggerOffline] Move Higgs sources in to plugins dir

cmsbuild committed:
Merge pull request #34916 from Dr15Jones/moveGuid

Move Guid.h to interface

cmsbuild committed:
Merge pull request #34912 from smorovic/120X-daq-start-from-ls

(DAQ) input source - minimum starting lumisection

cmsbuild committed:
Merge pull request #34710 from colizz/dev-121X-enableConcurrentGen

Enable concurrent GEN in Configuration/Generator/python

Dr15Jones committed:
Simplify setting EDPutTokenT via call to produces

Call to produces does not require specifying the data product type as it will be deduced via EDPutTokenT.

cmsbuild committed:
Merge pull request #34858 from slava77/patch-102

initialize Trackster::iterationIndex_

cmsbuild committed:
Merge pull request #34866 from smuzaffar/reco1-hdr-cleanup

[Reco] Removed unnecessary include statements

Commit:
Attempt to define unqiue names of materials everywhere

makortel committed:
Migrate SeedingLayerSetsBuilder to esConsumes()

Commit:
Correct the cfg's so that they can work

makortel committed:
Migrate PixelInactiveAreaFinder to esConsumes()

makortel committed:
Migrate DeepCoreSeedGenerator to esConsumes()

cmsbuild committed:
Merge pull request #34913 from smuzaffar/PhysicsTools-hdr-fix0817

[PhysicsTools] Move icc file used by public interface in to interface

cmsbuild committed:
Merge pull request #34863 from smuzaffar/L1TTrackMatch-hdr-fix

[L1TTrackMatch] Remove unnecessary include statements to avoid private hdr dep

makortel committed:
Migrate OutsideInMuonSeeder to esConsumes()

makortel committed:
Migrate SimpleCosmicBONSeeder to esConsumes()

fwyzard committed:
Fix radix sort test

Take into account the actual size of the type being sorted.

Use an union instead of a cast to keep the compiler happy about the
aliasing rules.

smuzaffar committed:
included MessageLogger.h LogDebug

cmsbuild committed:
Merge pull request #34928 from bsunanda/Run3-sim102

Run3-sim102 Correct the cfg's so that they can work

cmsbuild committed:
Merge pull request #34907 from smuzaffar/geom-hdr-fix0817

[Geometry] Move public headers in to interface dir

cmsbuild committed:
Merge pull request #34910 from makortel/esconsumesTrackClassifier

Complete migration of TrackClassifier to esConsumes()

Commit:
rename a file correctly

cmsbuild committed:
Merge pull request #34929 from fwyzard/fix_radix_sort_test_121x

Fix radix sort test

cmsbuild committed:
Merge pull request #34865 from smuzaffar/reco-hdr-cleanup

[Reco] Removed unnecessary include statements

makortel committed:
Migrate CosmicTrackFinder to esConsumes()

cmsbuild committed:
Merge pull request #34767 from iarspider/patch-6

[GCC 11] Fix error "this pointer is null" in L1Trigger/DTTrigger

makortel committed:
Migrate MeasurementTrackerEventProducer to esConsumes()

smuzaffar committed:
cleanup self include

makortel committed:
Migrate PFTauPrimaryVertexProducerBase to esConsumes()

smuzaffar committed:
remove self include

Dr15Jones committed:
Resolve compilation problem with clang

clang has a less permissive `using` implementation. Had to move all `produces` functions to base class.

makortel committed:
Remove unnecessary default constructor of SeedingLayerSetsBuilder

This allows declaring ESGetToken as const.

makortel committed:
Migrate PFSimParticleProducer to esConsumes()

makortel committed:
Migrate CSCHaloAlgo to esConsumes()

Commit:
Reaarange the files to remove .h files from SimCalorimetry/CastorSim/src

makortel committed:
Migrate PFMETAlgorithmMVA to esConsumes()

cmsbuild committed:
Merge pull request #34924 from makortel/esconsumesParametersDefinerTP

Migrate (Cosmic)ParametersDefinerForTP to esConsumes()

Commit:
Reaarange the files to remove .h files from SimCalorimetry/HcalSimProducers/src

Dr15Jones committed:
Better reporting of CPU timing

The Timing service reported CPU time for child processes as part of the total CPU time, but not as part of the loop time.
The timing summary did not account for child processes at all.

makortel committed:
Migrate PFMETProducer to esConsumes()

makortel committed:
Migrate EEBadScFilter to esConsumes()

makortel committed:
Migrate EcalBadCalibFilter to esConsumes()

makortel committed:
Migrate EcalDeadCellBoundaryEnergyFilter to esConsumes()

makortel committed:
Migrate EcalLaserCorrFilter to esConsumes()

Dr15Jones committed:
removed commented out code

cmsbuild committed:
Merge pull request #34843 from jshlee/GEM-unpacker-cleanup-CMSSW_12_1_0_pre1

GEM unpacker cleanup

makortel committed:
Migrate RPCRecHitProducer to esConsumes()

cmsbuild committed:
Merge pull request #34885 from mmusich/Alignment/CommonAlignmentProducer_migrateToConsumes

migrate `AlignmentProducer` to event consumes

cmsbuild committed:
Merge pull request #34899 from Dr15Jones/omtWorkerThread

Run G4 workers on their own dedicated threads

cmsbuild committed:
Merge pull request #34904 from smuzaffar/validation-hdr-fix0817

[Validation] Move EcalDigis sources in to plugins dir

cmsbuild committed:
Merge pull request #34872 from smuzaffar/IOMC-hdr-fix

[IOMC] Move public headers to interface directory

fwyzard committed:
Extend the cms::cuda::ScopedSetDevice interface

Add a default constructor that stores the current devices, without
changing it.

Add a set() method to change the current device, without affecting the
stored value for the original device.

fwyzard committed:
Fix uploading the EventSetup conditions to multiple CUDA devices

Associate to the correct CUDA device the events used to track if the
conditions have been transferred to each device.

mtosi committed:
fix typo in the gen fragment file name

abhih1 committed:
Swap x and y labels for a trigger plot in ECAL DQM

lecriste committed:
Remove unnecessary initialization

cmsbuild committed:
Merge pull request #34941 from Dr15Jones/betterCPUtiming

Better reporting of CPU timing

makortel committed:
Migrate JetMuonHitsIDHelper to esConsumes()

cmsbuild committed:
Merge pull request #34914 from Dr15Jones/GLBMuonAnalyzer

Consolidated GLBMuonAnalyzer into one file

fwyzard committed:
HLT customisation for Patatrack pixel tracks and GPU offload

Add a customisation script for the HLT:
  - enable the Patatrack pixel track reconstruction, with quadruplets or triplets, running on CPU or GPU
  - optionally, add the possibility to offload the ECAL unpacking and local reconstruction to run on GPU
  - optionally, add the possibility to offload the HCAL local reconstruction to run on GPU

Co-authored-by: Andrea Massironi <massironi.andrea@gmail.com>

cmsbuild committed:
Merge pull request #34908 from bainbrid/LowPtElectrons_SeedBDTs_XMLToRoot

LowPtElectrons: convert Seed BDTs from XML to ROOT file format

makortel committed:
Migrate PileupJetIdProducer to esConsumes()

cmsbuild committed:
Merge pull request #34951 from mtosi/fixBuJPsiPrimeK

fix typo in the gen fragment file name

makortel committed:
Add public interface for constructing and freeing caching allocators

makortel committed:
Migrate QGTragger to esConsumes

makortel committed:
Avoid bare 'new' in QGTagger

makortel committed:
Migrate JetTracksAssociatorAtCaloFace to esConsumes()

makortel committed:
Migrate TrackExtrapolator to esConsumes()

makortel committed:
Migrate HIPixelClusterVtxProducer to esConsumes()

cmsbuild committed:
Merge pull request #34953 from abhih1/ChangeTrigPlotLabel_Master

Correct labels for a Trigger plot in ECAL DQM

cmsbuild committed:
Merge pull request #34948 from fwyzard/CUDA_fix_concurrent_EventSetup_filling

Fix uploading the EventSetup conditions to multiple CUDA devices

cmsbuild committed:
Merge pull request #34954 from lecriste/parallelValidation

[HGCAL] Improve validation scripts

osschar committed:
Strip topolgy for sub-classes of StripTopology was not exported.

makortel committed:
Migrate HIProtoTrackFilterProducer to esConsumes()

osschar committed:
clang format

makortel committed:
Migrate HiEvtPlaneFlatProducer and EvtPlaneProducer to esConsumes()

makortel committed:
Migrate HiEgammaSCCorrectionMaker to esConsumes()

makortel committed:
Migrate HiSpikeCleaner to esConsumes()

makortel committed:
Remove useless try-catch's from HiSpikeCleaner

smuzaffar committed:
fix header guard and use full include path

slava77 committed:
initialize TrackstersProducer::iterIndex_ to 0

slava77 committed:
cast

cmsbuild committed:
Merge pull request #34956 from fwyzard/HLT_customisation_for_GPU_121x

HLT customisation for Patatrack pixel tracks and GPU offload

Commit:
Take care of Andrea's comments

Commit:
Take care of Andrea's comments

abhih1 committed:
Changes to fix ESConsumes

cmsbuild committed:
Merge pull request #34919 from smuzaffar/reco-hdr-fix0817

[Reco] Fixes needed to avoid exposing private headers via interface dir

abhih1 committed:
Fix code style

Commit:
Add tools to study big xml file which goes in the D

Dr15Jones committed:
Corrected comments

makortel committed:
Migrate TrackTimeValueMapProducer to esConsumes()

cmsbuild committed:
Merge pull request #34958 from makortel/cachingAllocatorHeader

Add public interface for constructing and freeing caching allocators

makortel committed:
Migrate TrackAssociatorByChi2Producer to esConsumes()

makortel committed:
Migrate TrackAssociatorByHitsProducer to esConsumes()

makortel committed:
Migrate TrackAssociatorByPositionProducer to esConsumes()

makortel committed:
Restore cms::cuda::ESProduct construction without GPU

makortel committed:
Migrate TrackingParticleNumberOfLayersProducer to esConsumes()

cmsbuild committed:
Merge pull request #34939 from bsunanda/Run3-hcx308

Run3-hcx308 Rearrange the files to remove .h files from SimCalorimetry/CastorSim/src

cmsbuild committed:
Merge pull request #34940 from bsunanda/Run3-hcx309

Run3-hcx309 Rearrange the files to remove .h files from SimCalorimetry/HcalSimProducers/src

cmsbuild committed:
Merge pull request #34960 from osschar/from-CMSSW_12_1_X_2021-08-18-1500

[Fireworks/Geometry] Strip topolgy for sub-classes of StripTopology was not exported to RECO geometry.

Commit:
Some changes made for testing big xml from DDD

Commit:
Add cfg's for DDD Big XML

cmsbuild committed:
Merge pull request #34935 from makortel/esconsumesRecoTauTag

Migrate PFTauPrimaryVertexProducerBase to esConsumes()

cmsbuild committed:
Merge pull request #34938 from makortel/esconsumesRecoParticleFlow

Migrate PFSimParticleProducer to esConsumes()

cmsbuild committed:
Merge pull request #34959 from makortel/esconsumesRecoJets

Migrate some modules in RecoJets to esConsumes()

cmsbuild committed:
Merge pull request #34961 from makortel/esconsumesRecoHI

Migrate some modules in RecoHI to esConsumes()

cmsbuild committed:
Merge pull request #34926 from Dr15Jones/simpler_produces

Simplify setting EDPutTokenT via call to produces

cmsbuild committed:
Merge pull request #34943 from makortel/esconsumesRecoMET

Migrate some modules in RecoMET to esConsumes()

cmsbuild committed:
Merge pull request #34934 from makortel/esconsumesRecoTrack

Migrate some modules in RecoTracker to esConsumes()

cmsbuild committed:
Merge pull request #34944 from makortel/cleanupTrackAssociatorRecord

Remove MuonToSimAssociatorBase, MuonToSimAssociatorByHits, and TrackAssociatorRecord as obsolete

cmsbuild committed:
Merge pull request #34962 from slava77/patch-103

initialize TrackstersProducer::iterIndex_ to 0

cmsbuild committed:
Merge pull request #34967 from makortel/esconsumesSimTracker

Migrate some modules in SimTracker to esConsumes()

cmsbuild committed:
Merge pull request #34969 from makortel/esconsumesTrackingParticleNumberOfLayersProducer

Migrate TrackingParticleNumberOfLayersProducer to esConsumes()

cmsbuild committed:
Merge pull request #34968 from makortel/cudaESProductWithoutGPU

Restore cms::cuda::ESProduct construction without GPU

Commit:
Put back HcalDigiProducer.cc in SimCalorimetry/HcalSimProducers/src

cmsbuild committed:
Merge pull request #34972 from bsunanda/Run3-hcx310

Run3-hcx310 Put back HcalDigiProducer.cc in SimCalorimetry/HcalSimProducers/src

cmsbuild committed:
Merge pull request #34927 from bsunanda/Run3-gex89

Run3-gex89 Attempt to define unqiue names of materials everywhere

cmsbuild committed:
Merge pull request #34964 from bsunanda/Run3-sim103

Run3-sim103 Add tools to study big xml file which goes in the DB

abhih1 committed:
Fix legacy module inherit warning

cmsbuild committed:
Merge pull request #34945 from makortel/esconsumesRPCRecHitProducer

Migrate RPCRecHitProducer to esConsumes()

Dr15Jones committed:
Avoid memory overwrites in Ecal PayloadInspectors

Use std::array to properly assign memory for multiple TPads.
This problem was found by gcc 11.

cmsbuild committed:
Merge pull request #34963 from abhih1/ESConsumesFixDBModule_master

Migrate leftover ECAL DQM codes to ESConsumes

cmsbuild committed:
Merge pull request #34922 from tvami/FixHCALrespAssymetry

Add HCAL tag `HcalRespCorrs_2021_v3.0_mc` to Run-3 MC GTs

cmsbuild committed:
Merge pull request #34982 from Dr15Jones/fixEcalPayloadInspectors

Avoid memory overwrites in Ecal PayloadInspectors

fwyzard commented:
Long term we should check if wecan speed up the code affected by `maxCellsPerHit`.
In the meantime could you check (or confirm) that we need to double it (from `128` to `256`), or if an intermediate value would be enough ?

Commit:
Merged CMSSW_12_1_X_GPU_HIon_v1 from repository stahlleiton with cms-merge-topic

Commit:
Updated the maxCellsPerHit threshold

stahlleiton commented:
Sorry, seems I made a mistake while trying to go to a different release (the daily CMSSW development release I was using had already expired).

How do I go to a newer release without all commits appearing in the history?

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-34811/24869

- This PR adds an extra 16KB to repository

- Found files with invalid states:
  - Geometry/VeryForwardData/data/dd4hep/cmsExtendedGeometry2021,xml:
    - Added: 75859ae9ffd3489ce2ef4b00fd38e6e2cc8fcd87
    - Deleted: e57130336cf4d828aed8bee80d2232a41956d67b
  - SimCalorimetry/HcalSimProducers/plugins/HcalDigiProducer.cc:
    - Added: bbdfbaf40bc2d46a25730cc51e93dbc3e864f87e
    - Deleted: fea5cc2537ed035789a44e4dac9a8e56e079edd3
  - SimCalorimetry/HcalSimProducers/plugins/SealModule.cc:
    - Deleted: bbdfbaf40bc2d46a25730cc51e93dbc3e864f87e
    - Added: fea5cc2537ed035789a44e4dac9a8e56e079edd3
  - Validation/Geometry/test/runMaterialBudgetVolumeBigXML_cfg.py:
    - Added: b04428ba000ae3ae495caa7fe4094368d15610c2
    - Deleted: ec08c83ccecdd3c3875e44bff8ffef7db4ee4be6
  - SimG4Core/PrintGeomInfo/data/dd4hep/hcalRecNumbering.xml:
    - Added: b04428ba000ae3ae495caa7fe4094368d15610c2
    - Deleted: 076def72d5ed387490aaa2cba1f179dbd745b890
  - SimCalorimetry/HcalSimProducers/src/HcalDigiProducer.cc:
    - Deleted: bbdfbaf40bc2d46a25730cc51e93dbc3e864f87e
    - Added: fea5cc2537ed035789a44e4dac9a8e56e079edd3
  - SimG4Core/PrintGeomInfo/data/dd4hep/hcalSimNumbering.xml:
    - Added: b04428ba000ae3ae495caa7fe4094368d15610c2
    - Deleted: 076def72d5ed387490aaa2cba1f179dbd745b890

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Alignment/OfflineValidation/python/TkAlAllInOneTool/offlineValidationTemplates.py  modified in PR(s): #35016
  - File  CondCore/Utilities/src/CondDBFetch.cc  modified in PR(s): #34932
  - File  Configuration/Applications/python/ConfigBuilder.py  modified in PR(s): #34985
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #34714, #34985
  - File  Configuration/StandardSequences/python/AlCaRecoStreams_cff.py  modified in PR(s): #34985
  - File  DQM/SiPixelMonitorDigi/src/SiPixelDigiSource.cc  modified in PR(s): #32875
  - File  FWCore/Framework/src/EDAnalyzer.cc  modified in PR(s): #31746
  - File  FWCore/Framework/src/EDProducer.cc  modified in PR(s): #31746
  - File  FWCore/Framework/src/global/EDAnalyzerBase.cc  modified in PR(s): #31746
  - File  FWCore/Framework/src/global/EDFilterBase.cc  modified in PR(s): #31746
  - File  FWCore/Framework/src/global/EDProducerBase.cc  modified in PR(s): #31746
  - File  FWCore/Framework/src/limited/EDAnalyzerBase.cc  modified in PR(s): #31746
  - File  FWCore/Framework/src/limited/EDFilterBase.cc  modified in PR(s): #31746
  - File  FWCore/Framework/src/limited/EDProducerBase.cc  modified in PR(s): #31746
  - File  FWCore/Framework/src/one/EDAnalyzerBase.cc  modified in PR(s): #31746
  - File  FWCore/Framework/src/one/EDFilterBase.cc  modified in PR(s): #31746
  - File  FWCore/Framework/src/one/EDProducerBase.cc  modified in PR(s): #31746
  - File  FWCore/Framework/src/stream/EDAnalyzerAdaptorBase.cc  modified in PR(s): #31746
  - File  FWCore/Framework/src/stream/EDFilterAdaptorBase.cc  modified in PR(s): #31746
  - File  FWCore/Framework/src/stream/EDProducerAdaptorBase.cc  modified in PR(s): #31746
  - File  FWCore/Framework/src/stream/ProducingModuleAdaptorBase.cc  modified in PR(s): #31746
  - File  FWCore/Framework/test/BuildFile.xml  modified in PR(s): #31746
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #34978, #34788
  - File  Mixing/Base/src/PileUp.cc  modified in PR(s): #34902
  - File  PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py  modified in PR(s): #34860, #33885
  - File  SimG4Core/PrintGeomInfo/test/SimFileCompare.cpp  modified in PR(s): #35015
  - File  SimPPS/PPSSimTrackProducer/plugins/PPSSimTrackProducer.cc  modified in PR(s): #34989
  - File  Validation/EcalDigis/plugins/EcalMixingModuleValidation.cc  modified in PR(s): #34902
  - File  Validation/EcalDigis/plugins/EcalMixingModuleValidation.h  modified in PR(s): #34902
  - File  Validation/Geometry/macros/MatBudgetVolume.C  modified in PR(s): #35015
  - File  Validation/RecoParticleFlow/test/compare.py  modified in PR(s): #34580

cmsbuild commented:
Pull request #34811 was updated. @SiewYan, @andrius-k, @jpata, @lveldere, @chayanit, @bbilin, @wajidalikhan, @sbein, @ianna, @Saptaparna, @Martin-Grunewald, @rekovic, @tlampen, @alberto-sanchez, @pohsun, @santocch, @cecilecaillol, @perrotta, @civanch, @yuanchao, @makortel, @ErnestaP, @ahmad3213, @cmsbuild, @agrohsje, @fwyzard, @GurpreetSinghChahal, @smorovic, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @mdhildreth, @jfernan2, @kpedro88, @kskovpen, @slava77, @ggovi, @qliphy, @fabiocos, @francescobrivio, @malbouis, @ssekmen, @mkirsano, @jordan-martins, @adrianodee, @kmaeshima, @emeschi, @alja, @srimanob, @rvenditti, @tvami can you please check and sign again.
